src/invocation/ClientConnectionManager.ts

```diff
@@ -95,14 +96,10 @@ class ClientConnectionManager extends EventEmitter {
         this.emit(EMIT_CONNECTION_OPENED, connection);
     }
 
-    private authenticate(connection: ClientConnection): Q.Promise<boolean> {
-        var name = this.client.getConfig().groupConfig.name;
-        var password = this.client.getConfig().groupConfig.password;
-        var invocationService = this.client.getInvocationService();
-
-        var authenticator = new ConnectionAuthenticator(connection, invocationService, name, password);
+    private authenticate(connection: ClientConnection, ownerConnection: boolean): Q.Promise<boolean> {
+        var authenticator = new ConnectionAuthenticator(connection, this.client);
```

Comment:
You have to create a constructor variable, and then a class field -> too much typing without appreciable benefit :)